lib/api/user_token_service.rb

```diff
@@ -3,21 +3,25 @@ class UserTokenService
     def initialize(config = Settings, args = {})
       @config = config
       @svc_options = args
-      @token_mgr = new_token_mgr(base_config[:module], base_config[:name], api_config)
     end
 
-    attr_accessor :token_mgr
+    def token_mgr(namespace = nil)
+      @token_mgr ||= {}
+      if namespace.nil? || namespace == 'ui' # The default token and UI token share the same TokenStore namespace
+        @token_mgr['ui'] ||= new_token_mgr(base_config[:module], base_config[:name], api_config)
+      elsif namespace == 'ws'
+        @token_mgr['ws'] ||= TokenManager.new('ws', :token_ttl => 5.minutes.to_i)
```

Comment:
Discussing this with @skateman right now. It turns out that the WebSocket may disconnect/reconnect, hence we need long lived token. In that case it makes sense for the token to live as long as the ui token, hence `::Settings.session.timeout` seems sensible. Will amend.


Why there was a 5 minutes before?

In previous discussions we had an idea to make the token short-lived, but then we went with totally separated tokens. Given the risks (i.e. the token for WS needs to be stored in a cookie by client. Since browsers do not apply same-origin cookie policy on WS, the cookie can be hijacked. Hence, we want to have WS token separate. Then, we can mitigate cross-site WS hijacking by having the server validate 'Origin' header). So, now we have other mechanisms to secure the tokens and we don't need the band-aid of having the tokens short-lived.